### PR TITLE
OZ-573: Add support to authenticate with `oauth2` via camel route

### PR DESCRIPTION
**Pull request by kipchubett**
Issue: https://mekomsolutions.atlassian.net/browse/OZ-573

This PR adds OAuth2 authentication support

### CODE REVIEW

File: commons/src/main/java/com/ozonehis/eip/security/oauth2/Oauth2Processor.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import org.apache.camel.Processor;
+import org.apache.camel.ProducerTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Setter
+@Getter
+@Component("eip.oauthProcessor")
+public class Oauth2Processor implements Processor {
```
Comment (VaishSiddharth) [inline commons/src/main/java/com/ozonehis/eip/security/oauth2/Oauth2Processor.java]: It should be `OAuth2Processor`
Same for `Oauth2Route` `Oauth2Token` and test files

File: commons/src/main/java/com/ozonehis/eip/security/oauth2/Oauth2Processor.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    private final ProducerTemplate producerTemplate;
+
+    @Autowired
```
Comment (VaishSiddharth) [inline commons/src/main/java/com/ozonehis/eip/security/oauth2/Oauth2Processor.java]: Do we need this annotation here?

**Comment by VaishSiddharth** [comment]
@corneliouzbett Left some comments otherwise looks good to me.

File: commons/src/main/java/com/ozonehis/eip/security/oauth2/Oauth2Processor.java (hunk 2 above)
Comment (kipchubett) [reply commons/src/main/java/com/ozonehis/eip/security/oauth2/Oauth2Processor.java]: Yes, we need it to inject `producerTemplate` Spring bean.

**Review by VaishSiddharth — APPROVED** [review]
LGTM!

**Pull request merged**